WebApp/handler/SystemAdministrationHandler.py

```diff
@@ -22,83 +21,41 @@ class SystemAdministrationHandler(WebHandler):
-                                           delegatedGroup=group)
-        resultHost = yield self.threadTask(client.getHostInfo)
-        if resultHost["OK"]:
-          resultHost["Value"]["Host"] = host
-          if "Timestamp" in resultHost["Value"]:
-            resultHost["Value"]["Timestamp"] = str(resultHost["Value"]["Timestamp"])
-          callback.append(resultHost["Value"])
-        else:
-          callback.append({"Host": host})
-
-    total = len(callback)
-    if not total > 0:
-      self.finish({"success": "false", "error": "No system information found"})
+    client = ComponentMonitoringClient(delegatedDN=self.getUserDN(), delegatedGroup=self.getUserGroup())
+    result = yield self.threadTask(client.getLogs)
+    if not result['OK'] or not len(result['Value']) > 0:
+      self.finish({"success": "false", "error": result.get('Message', "No system information found")})
       return
 
-    callback = sorted(callback, key=lambda i: i['Host'])
+    callback = result['Value']
 
     # Add the information about the extensions' versions, if available, to display along with the DIRAC version
     for i in range(len(callback)):
-      if 'Extension' in callback[i]:
-        # We have to keep the backward compatibility (this can heppen when we do not update one host to v6r15 ...
-        callback[i]['DIRAC'] = '%s,%s' % (
-            callback[i].get('DIRACVersion', callback[i].get('DIRAC')), callback[i]['Extension'])
-      elif 'Extensions' in callback[i]:
-        callback[i]['DIRAC'] = '%s,%s' % (callback[i].get('DIRAC', callback[i].get('DIRAC')), callback[i]['Extensions'])
+      callback[i]["Host"] = callback[i]["HostName"]
+      callback[i]["Timestamp"] = str(callback[i].get("Timestamp", "unknown"))
+      # We have to keep the backward compatibility (this can heppen when we do not update one host to v6r15 ...
+      callback[i]['DIRAC'] = '%s,%s' % (callback[i].get('DIRACVersion', callback[i].get('DIRAC', '')),
+                                        callback[i].get('Extension', callback[i].get('Extensions', '')))
 
-    self.finish({"success": "true", "result": callback, "total": total})
+    self.finish({"success": "true", "result": sorted(callback, key=lambda i: i['Host']), "total": len(callback)})
 
   @asyncGen
   def web_getHostData(self):
     """
     Returns flatten list of components (services, agents) installed on hosts
     returned by getHosts function
     """
-    DN = self.getUserDN()
-    group = self.getUserGroup()
-
     callback = list()
 
     if "hostname" not in self.request.arguments and not self.request.arguments["hostname"][0]:
       self.finish({"success": "false", "error": "Name of the host is absent"})
       return
 
     host = self.request.arguments["hostname"][0]
-    client = SystemAdministratorClient(host, None, delegatedDN=DN,
-                                       delegatedGroup=group)
+    client = SystemAdministratorClient(host, None, delegatedDN=self.getUserDN(),
```

Comment:
I think this does not work as the self.getUserDN() will be resolved at the moment of the code loading.
```suggestion
    userDN = self.getUserDN()
    client = SystemAdministratorClient(host, None, delegatedDN=userDN(),
```
and the same for the group

Reply:
returned as it was before